### ISSUE TITLE
fix output sequence

### ISSUE THREAD

**Issue by adnjoo** [adnjoo/pytorch-transformer#8]
It looks like your model has completed training, but the output sequence prediction consists mainly of [unused] tokens and repeated subword tokens like ##ann. This behavior suggests a few potential issues:

1. Vocabulary and Tokenization Issue
The repeated [unused] tokens and subword fragments (##ann) could indicate that the model is not properly aligned with the tokenizer or vocabulary. Here are a few things to check:

Vocabulary Mismatch: Ensure that the vocabulary (vocab.json) used for tokenizing the input and decoding the output is the same one used during training. If these are out of sync, the model might predict token indices that map to unused tokens.

Proper Tokenization: Verify that your input text is correctly tokenized using the same tokenizer that was used during training. If not, the model might receive input in an unexpected format, resulting in meaningless output.

2. Training Issues
The loss during training decreases, but the model may not have fully converged. If the loss is still quite high, the model might not have learned useful representations yet. Some things to try:

Longer Training: Train for more epochs. The model might need more iterations to better learn the task. Alternatively, consider adjusting the learning rate or other hyperparameters.

More Data: If you're working with a small dataset, the model might not have enough data to learn properly. Augmenting your dataset or using a pretrained model as a starting point might help.

3. Checkpoint Loading
The warning you're seeing about torch.load is related to the security of loading model files. To avoid security risks, you should set weights_only=True when loading a checkpoint. You can modify the code where the checkpoint is loaded like this:

python
Copy code
model.load_state_dict(torch.load(checkpoint_path, weights_only=True))
This change ensures that only the model weights are loaded, avoiding potential issues with executing arbitrary code.

4. Unused Tokens in the Vocabulary
The [unused] tokens may be placeholders in the vocabulary. These are common in certain models (e.g., BERT) to reserve space for future tokens or special tokens. If your model is predicting many of these, it may not have learned to output meaningful tokens yet. Try:

Fine-tuning: If this is a pre-trained model, it might require more fine-tuning on your specific data to start making meaningful predictions.
Suggestions:
Debug the Tokenizer: Check that the tokenizer and vocabulary are consistent between training and inference.
Train for More Epochs: The model might need more training or better hyperparameters.
Address torch.load Warning: Use weights_only=True to ensure safe loading of checkpoints.
Let me know if you need further help debugging!
